docs/DeveloperGuide.md

```diff
@@ -1,34 +1,429 @@
+one location to another. In finding the shortest route, it utilises the breath-first-search algorithm, which will be 
+further elaborated in the implementation section.
+
+As shown in the diagram above, `Router` is used by the following classes:
+* `GoCommand`
+* `RepeatHistoryCommand`
+* `RepeatFavouriteCommand`
+* `ShowDailyRouteCommand`
+
+### 2.6. Data Component
+![img.png](images/DataComponent.png)
+
+The **Data Component** is where all the data that are needed to execute a command is stored. For example when `"go"`
+command is executed, the `GoCommand`object will use data stored in `NusMap`, `EateryList` and `BlockAlias` in order to find
+the shortest route.
+
+On the other hand, `Storage` is responsible for saving and loading data stored in the **Data Component**. This will be
+further elaborated in the following section.
+
+### 2.7. Storage Component  
+
+---------------------------------------------------------------------------------------------
+
+## *3. Implementation*
+### 3.1. Save feature
+#### Current Implementation
+The save mechanism is facilitated by `AliasStorage`, `HistoryRouteStorage`, `NotesStorage`, `DailyRouteStorage`, `FavouriteLocationsStorage` subclasses. </br>
+They extend `Storage` (superclass) with a feature to save the block aliases, history of visited routes, tagged notes, daily routes and favourite locations, stored internally as a `aliasList`,  `historyList`, `notesList`, `dailyRouteList` and `favouritesList`. <br />
+Additionally, they implement the following operations: <br/>
+- `AliasStorage#overwriteAliasListFile()` —  Saves all aliases given by user to blocks into `aliasList`. <br />
+- `AliasStorage#loadAlias()`   —  Restores all aliases given by user to blocks from `aliasList`. <br />
+- `HistoryRouteStorage#overwriteHistoryListFile()` —  Saves the current list of the 10 most recently visited routes in its history into `historyList`. <br />
+- `HistoryRouteStorage#loadHistory()` —  Restores the previous list of the 10 most recently visited routes in its history from `historyList`. <br />
+- `NotesStorage#overwriteNotesListFile()` —  Saves all notes tagged to a location into `notesList`. <br />
+- `NotesStorage#loadNotes()` —  Restores all notes tagged to a location from `notesList`. <br />
+- `DailyRouteStorage#loadDailyRoute()` —  Saves all the daily routes that user wants to see for each day of the week into `dailyRouteList`. <br />
+- `DailyRouteStorage#overwriteDailyRouteFile()` Restores all the daily routes that user wants to see from `dailyRouteList`. <br />
+- `FavouriteLocationsStorage#overwriteFavouritesListFile()` —  Saves the current list of all the locations that the users are interested in keeping in `favouritesList`. <br />
+- `FavouriteLocationsStorage#loadFavourites()` —  Restores the previous list of the all the locations that the users are interested in keeping from `favouritesList`. <br />
```

Comment:
Consider simplifying the explanation of this feature, not sure if all the methods mentioned here are required in the explanation of the feature.